crates/bevy_reflect/src/path.rs

```diff
@@ -174,6 +174,26 @@ impl FieldPath {
         }
         Ok(current)
     }
+
+    pub fn get_field<'r, 'p, T: Reflect>(
+        &'p self,
+        root: &'r dyn Reflect,
+    ) -> Result<&'r T, ReflectPathError<'p>> {
+        self.field(root).and_then(|p| {
+            p.downcast_ref::<T>()
+                .ok_or(ReflectPathError::InvalidDowncast)
+        })
+    }
+
+    pub fn get_field_mut<'r, 'p, T: Reflect>(
+        &'p mut self,
+        root: &'r mut dyn Reflect,
+    ) -> Result<&'r mut T, ReflectPathError<'p>> {
+        self.field_mut(root).and_then(|p| {
+            p.downcast_mut::<T>()
+                .ok_or(ReflectPathError::InvalidDowncast)
+        })
+    }
```

Comment:
Would it make sense to add a method that consumes the `root` and returns a `Box<T>` or `T`?

Probably not, right? Since you likely never need just a single field.

Reply:
The primary envisioned use case for this was animation for every frame. This would be a matter of mutation via reference than extraction.

Dynamic path based extraction seems odd to me, I can't really think of a solid use case.

Reply:
Yeah me neither. Just curious if there was a use for it that I couldn't think of.